fplll/enum/enumerate_base.cpp

```diff
@@ -190,7 +190,7 @@ template <bool dualenum, bool findsubsols, bool enable_reset> void EnumerationBa
 
   partdist[k_end] = 0.0;  // needed to make next_pos_up() work properly
 
-  nodes -= k_end - k;
+  nodes[k_end] -= k_end - k;
```

Comment:
It seems there could be an overflow? This may affect the following?

      return std::accumulate(nodes.begin(), nodes.end(), 0);

Reply:
So I asked Marc about this. This code is apparently to retain compatibility with the loop below it (i.e if we decide to use the iterative enumeration vs the recursive enumeration). Whilst there will be an overflow, the recursive dispatch accounts for this, and so it will still return the correct count.

Reply:
I am puzzled. Just tried the test_counter.cpp. I added a line
    cout << "Level " << i << " counts " << a[i] << endl;
right after Line 28. 

For the first (10 + 1) levels, it prints:

Level 0 counts 1
Level 1 counts 1
Level 2 counts 1
Level 3 counts 1
Level 4 counts 1
Level 5 counts 1
Level 6 counts 4
Level 7 counts 4
Level 8 counts 3
Level 9 counts 2
Level 10 counts 18446744073709551606

Do you mean the Level 9 #nodes should be 12 instead of 2?

Reply:
Marc & Joe, the maximum mis-counting is by a value of k? If so, this is minor in practice for larger k.

Reply:
Whilst that's true, I think we should make sure that all of the enumerators return the same number of nodes. Otherwise, we could end up with a scenario in future where something needs to be debugged, and if the number of nodes is significantly different then that may cause some headaches. Of course for large k this 'significantly different' will be small compared to the overall count, but I'm still concerned about it in practice for small values of k.